### PR TITLE
AF-648 :  Change Project from a folder in a repo to a repository

### PR DESCRIPTION
**Pull request by ederign**
AF-648 :  Change Project from a folder in a repo to a repository
-requested changes

temp

tmp

safe guard for NPEs

last fixes ( I hope )

Dashbuilder issue on login

Loading is now showed until the end of project creation

Fix double click when selecting a space and keep assets open unless another project is selected

Fixed loading when opening project.

[AF-907] Organization Unit Migration for Spaces

Fixed import projects

AF-648: Create command line migration tool for workspace projects.

* Still need to solve packaging (weld-se does not work well in an uber-jar).
* Need to fix log configuration to avoid spamming console output.

AF-648: Package zip file for project migration tool with dependencies bundled and shell script to execute.

Don't try to migrate niogit that looks like it's already updated.

Cache with spaces

Fixed refresh assets list when going back to project screen after asset creation

Busy popup while loading spaces list.

Remove project migration from workbench UI.

Refactor + add abstractions over System/Files to allow unit testing.

Remove repo explicitly using space to avoid errors from no active context.

Silence console log messages in project migration tool.

Cleanup console output in project migration tool.

Suppress errors during Weld shutdown.

fixing broken tests

merge

forgot one file

Fix enforcer error in migration tool.

Remove WorkspaceProjectContext from backend use. Pass space as parameter.

Fixed breadcrumbs

### CODE REVIEW

File: kie-wb-common-cli/kie-wb-common-project-migration/src/main/sh/migration-tool.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+export CLASSPATH="`find lib -name '*.jar' | tr '\n' ':' | sed -E 's/:$//'`"
```
Comment (jhrcek) [inline kie-wb-common-cli/kie-wb-common-project-migration/src/main/sh/migration-tool.sh]: @mbarkley could we make this script independent on where it is executed from?
After building this repo I unzipped the zip with this tool and got error like this

```
cd kie-wb-common-project-migration-7.6.0-SNAPSHOT/bin
./migration-tool.sh

find: ‘lib’: No such file or directory
Error: Could not find or load main class org.kie.workbench.common.project.MigrationTool
```

I modified it like this, which made it possible to run from bin dir as well as from any other dir (like `kie-wb-common-project-migration-7.6.0-SNAPSHOT/bin/migration-tool.sh`

```
#!/bin/bash
SCRIPTDIR=$(dirname "$0")
CLASSPATH=$(find "${SCRIPTDIR}"/../lib -name '*.jar' | tr '\n' ':' | sed -E 's/:$//')
export CLASSPATH
java org.kie.workbench.common.project.MigrationTool "$@"
```

File: kie-wb-common-cli/kie-wb-common-project-migration/src/main/java/org/kie/workbench/common/project/cli/MigrationConstants.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.workbench.common.project.cli;
+
+
+public interface MigrationConstants {
+
+    // TODO Find where these constants come from
```
[the remark this answers is not in the document]
Comment (mbarkley) [reply kie-wb-common-cli/kie-wb-common-project-migration/src/main/java/org/kie/workbench/common/project/cli/MigrationConstants.java]: Not yet. Unfortunately most of these words (system, datasets, datasource, etc.) turn up many places in our code base, making it hard to search for them.

File: kie-wb-common-cli/kie-wb-common-project-migration/src/main/sh/migration-tool.sh (hunk 1 above)
Comment (mbarkley) [reply kie-wb-common-cli/kie-wb-common-project-migration/src/main/sh/migration-tool.sh]: Good idea. I'll incorporate this change.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/util/ResourceUtils.java [hunk 3]
```diff
@@ -58,7 +58,7 @@ public static boolean isProjectHandler(final NewResourceHandler handler) {
     }
 
     public static boolean isDefaultProjectHandler(final NewResourceHandler handler) {
-        return handler.getClass().getName().contains("org.kie.workbench.common.screens.projecteditor.client.handlers.NewProjectHandler");
+        return handler.getClass().getName().contains("org.kie.workbench.common.screens.projecteditor.client.handlers.NewWorkspaceProjectHandler");
```
Comment (jhrcek) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/util/ResourceUtils.java]: I'm not sure if it's possible in gwt client code, but wouldn't `NewWorkspaceProjectHandler.class.getName()` be more robust to future changes here?

File: kie-wb-common-cli/kie-wb-common-project-migration/src/main/java/org/kie/workbench/common/project/MigrationTool.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        SystemAccess system = new RealSystemAccess();
+        new MigrationTool(new ExternalMigrationService(system),
+                          new ToolConfig.DefaultFactory(),
+                          system,
+                          () -> {
+                            /*
+                             * Work around log4j-api message printed from missing configuration.
+                             */
+                            PrintStream err = System.err;
+//                            System.setErr(new PrintStream(new NullOutputStream()));
```
Comment (mbarkley) [inline kie-wb-common-cli/kie-wb-common-project-migration/src/main/java/org/kie/workbench/common/project/MigrationTool.java]: @Rikkola I see from the old AF-648 branch that you commented this (and related dependencies) out. Why?

Comment (Rikkola) [reply kie-wb-common-cli/kie-wb-common-project-migration/src/main/java/org/kie/workbench/common/project/MigrationTool.java]: @mbarkley I just temporally took it out to make a build work. Looks like this change has slipped in.

**Comment by mbarkley** [comment]
@ederign @Rikkola @paulovmr Force pushed to rebase off of master.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by ederign** [comment]
@mbarkley @paulovmr @Rikkola Force pushed to rebase off master.

**Comment by ederign** [comment]
Jenkins, please retest this.

**Comment by ederign** [comment]
Jenkins, please retest this.

**Comment by porcelli** [comment]
thank you also @tomasdavidorg @jhrcek for QE support

**Pull request merged**